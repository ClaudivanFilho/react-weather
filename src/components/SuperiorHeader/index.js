import React, { Component, PropTypes } from 'react';

import CityInput from '../CityInput'

import './index.scss';

export default class SuperiorHeader extends Component {
  
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="col-sm-12 app-pickup-form">
        { this.props.weather && this.props.weather.country &&
          <h2 className="float-left" style={{marginTop: "10px"}}>
            {this.props.weather.name + ', ' + this.props.weather.country}
          </h2> 
        }
        <CityInput />
      </div>
    );
  }
}

