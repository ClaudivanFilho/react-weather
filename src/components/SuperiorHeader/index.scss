.app-pickup-form {
  margin-top: 20px !important;

  h2 {
    margin-top: 10px;
  }
}
