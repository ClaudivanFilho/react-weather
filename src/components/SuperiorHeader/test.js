import React from 'react';
import {shallow} from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import SuperiorHeader from './index';

configure({ adapter: new Adapter() });

const weatherMock = {
  country: 'Br',
  name: 'Campina Grande'
}

const predictionMock = {
  date: '131414020',
  icon: '10n',
  info: 'Clouds'
}

test('Test SuperiorHeader Component', () => {
  const component = shallow(<SuperiorHeader weather={weatherMock}/>);
  expect(component.find('.app-pickup-form').length).toEqual(1);

  expect(toJson(component)).toMatchSnapshot();
});
