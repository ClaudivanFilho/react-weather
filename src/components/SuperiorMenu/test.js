import React from 'react';
import {shallow} from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import SuperiorHeader from './index';

configure({ adapter: new Adapter() });

test('Test SuperiorHeader Component', () => {
  const component = shallow(<SuperiorHeader />);
  expect(component.find('Link').length).toEqual(2);
  expect(component.find('div').length).toEqual(3);
  expect(toJson(component)).toMatchSnapshot();
});