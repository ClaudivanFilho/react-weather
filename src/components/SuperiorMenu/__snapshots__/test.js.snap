// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SuperiorHeader Component 1`] = `
<div
  className="ui fixed menu"
>
  <Link
    replace={false}
    to="/"
  >
    <div
      className="active item"
    >
      Current Weather
    </div>
  </Link>
  <Link
    replace={false}
    to="/forecast"
  >
    <div
      className=" item"
    >
      Forecast
    </div>
  </Link>
</div>
`;
